Plugins/BuildServerIntegration/TeamCityIntegration/TeamCityAdapter.cs

```diff
@@ -75,7 +75,7 @@ private CookieContainer GetTeamCityNtlmAuthCookie (string serverUrl, IBuildServe
 
             string url = serverUrl + "ntlmLogin.html";
             var cookieContainer = new CookieContainer();
-            var request = (HttpWebRequest)HttpWebRequest.Create (url);
+            var request = (HttpWebRequest)WebRequest.Create (url);
```

Comment:
I don't understand this change. How WebRequest can be cast to HttpWebRequest?

Reply:
There is no actual change to the generated IL here. Class `HttpWebRequest` derives from `WebReqeust`, but the method is static, so not overridden. This change results in the method being referenced via the type that actually declares it.

The method's return type depends upon the URI it's given. If you give it an `http://` URI, it'll return `HttpWebRequest`.

Reply:
Thanks, I code a lot in Pascal, so I unconsciously interpreted Create as a constructor :)